src/components/relation-control/index.js

```
@@ -2,378 +2,401 @@
+		updateBlockAttributes,
 		__unstableMarkNextChangeAsNotPersistent: markNextChangeAsNotPersistent,
 	} = useDispatch('core/block-editor');
 
-	const repeaterContext = useContext(RepeaterContext);
+	// UseRef to prevent infinite loops during attribute updates
+	const isUpdating = useRef(false);
+
+	// Track highlighted blocks for cleanup
+	const highlightedBlocks = useRef(new Set());
+	const hoveredUniqueIdRef = useRef(null);
+	const onChangeRef = useRef(onChange);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)